xwrf/io_plugin.py

```diff
@@ -51,17 +58,32 @@ def clean(dataset):
                 )
             except:
                 warnings.warn(f'Failed to parse time coordinate: {coord}', stacklevel=2)
-
         elif coord in (_LON_COORDS + _LAT_COORDS) and dataset[coord].ndim == 3:
-
-            attrs = dataset[coord].attrs
-            encoding = dataset[coord].encoding
             dataset = dataset.assign_coords(
                 {coord: (dataset[coord].dims[1:], dataset[coord].data[0, :, :])}
             )
+        elif coord in (_VERTICAL_COORDS) and dataset[coord].ndim == 2:
+            dataset = dataset.assign_coords(
+                {coord: (dataset[coord].dims[1:], dataset[coord].data[0, :])}
+            )
+
         dataset[coord].attrs = attrs
         dataset[coord].encoding = encoding
 
+    # Handle horizontal projection coordinates
+    dataset = add_horizontal_projection_coordinates(dataset)
+
+    # Clean up vertical and time coordinates (to make dimension coordinate) if able
+    if 'ZNU' in dataset.coords and 'bottom_top' in dataset.dims and dataset['ZNU'].ndim == 1:
+        dataset['bottom_top'] = dataset['ZNU']
+        del dataset['ZNU']
+    if 'ZNW' in dataset.coords and 'bottom_top_stag' in dataset.dims and dataset['ZNW'].ndim == 1:
+        dataset['bottom_top_stag'] = dataset['ZNW']
+        del dataset['ZNW']
```

Comment:
I wouls suggest adding XGCM attributes to `bottom_top` and `bottom_top_stag` like this:

```suggestion
    if 'ZNU' in dataset.coords and 'bottom_top' in dataset.dims and dataset['ZNU'].ndim == 1:                                                                                                                                                                 
        dataset['bottom_top'] = (                                                                                                                                                                                                                             
            'bottom_top',                                                                                                                                                                                                                                     
            dataset['ZNU'].values,                                                                                                                                                                                                                            
            {'standard_name': 'hybrid_vertical_coordinate', 'axis': 'Z'},                                                                                                                                                                                     
        )                                                                                                                                                                                                                                                     
        del dataset['ZNU']                                                                                                                                                                                                                                    
    if 'ZNW' in dataset.coords and 'bottom_top_stag' in dataset.dims and dataset['ZNW'].ndim == 1:                                                                                                                                                            
        dataset['bottom_top_stag'] = (                                                                                                                                                                                                                        
            'bottom_top_stag',                                                                                                                                                                                                                                
            dataset['ZNW'].values,                                                                                                                                                                                                                            
            {                                                                                                                                                                                                                                                 
                'standard_name': 'hybrid_vertical_coordinate',                                                                                                                                                                                                
                'axis': 'Z',                                                                                                                                                                                                                                  
                'c_grid_axis_shift': 0.5,                                                                                                                                                                                                                     
            },                                                                                                                                                                                                                                                
        )                                                                                                                                                                                                                                                     
        del dataset['ZNW']                                                                                                                                                                                                                                    
```